hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-applicationhistoryservice/src/main/java/org/apache/hadoop/yarn/server/timeline/util/LeveldbUtils.java

```diff
@@ -203,6 +207,8 @@ public static DB loadOrRepairLevelDb(JniDBFactory factory, Path dbPath, Options
       File dbFile = new File(dbPath.toString());
       File dbBackupPath = new File(
           dbPath.toString() + BACKUP_EXT + Time.monotonicNow());
+      LOG.warn("Incurred exception while loading LevelDb database. Backing " +
+          "up at "+ dbBackupPath, ioe);
```

Comment:
Would you use slf4j placeholder (`{}`) for logging? It enables lazy-evaluation and improve performance especially in debug log. http://www.slf4j.org/faq.html#logging_performance

Reply:
Oh, the previous code did not use the parameterized logging.